src/modules/ekf2/EKF/mag_control.cpp

```diff
@@ -210,12 +210,16 @@ void Ekf::controlMagFusion()
 
 			if (_mag_yaw_reset_req || !_control_status.flags.yaw_align || mag_sample.reset || !mag_enabled_previously || haglYawResetReq()) {
 
-				if (magReset()) {
+				if ((_mag_counter > 1) && resetMagStates(_mag_lpf.getState())) {
```

Comment:
For another PR but we might want to give a bit more samples than just 1 to consider it a "filtered value"

Reply:
Yep